boot/bootutil/include/bootutil/crypto/ecdsa.h

```diff
@@ -0,0 +1,585 @@
+    sig_ptr  = &sig_ptr[r_len+2];
+    if (s_len >= num_of_curve_bytes) {
+        sig_ptr = sig_ptr + s_len - num_of_curve_bytes;
+        memcpy(&r_s_pair[num_of_curve_bytes], sig_ptr, num_of_curve_bytes);
+    } else {
+        /* For encodings that reduce the size of r or s in case of zeros */
+        memcpy(&r_s_pair[2*num_of_curve_bytes - s_len], sig_ptr, s_len);
+    }
+}
+
+// OID id-ecPublicKey 1.2.840.10045.2.1.
+const uint8_t IdEcPublicKey[] = {0x06, 0x07, 0x2a, 0x86, 0x48, 0xce, 0x3d, 0x02, 0x01};
+// OID secp256r1 1.2.840.10045.3.1.7.
+const uint8_t Secp256r1[] = {0x06, 0x08, 0x2a, 0x86, 0x48, 0xce, 0x3d, 0x03, 0x01, 0x07};
+// OID secp384r1 1.3.132.0.34
+const uint8_t Secp384r1[] = {0x06, 0x05, 0x2b, 0x81, 0x04, 0x00, 0x22};
+
+static inline void bootutil_ecdsa_init(bootutil_ecdsa_context *ctx)
+{
+    ctx->key_id = PSA_KEY_ID_NULL;
+    ctx->curve_byte_count = 0;
+    ctx->required_algorithm = 0;
+}
+
+static inline void bootutil_ecdsa_drop(bootutil_ecdsa_context *ctx)
+{
+    if (ctx->key_id != PSA_KEY_ID_NULL) {
+        (void)psa_destroy_key(ctx->key_id);
+    }
+}
+
+/*
+ * Parse a ECDSA public key with format specified in RFC5280 et al.
+ *
+ * OID for icEcPublicKey is 1.2.840.10045.2.1
+ * OIDs for supported curves are as follows:
+ *     secp256r1 (prime256v1): 1.2.840.10045.3.1.7
+ *     secp384r1: 1.3.132.0.34
+ */
+static inline int bootutil_ecdsa_parse_public_key(bootutil_ecdsa_context *ctx,
+                                                       uint8_t **cp, uint8_t *end)
+{
+    psa_key_attributes_t key_attributes = psa_key_attributes_init();
+    size_t key_size;
+    (void)end;
+
+    /* public key oid is valid */
+    if (memcmp(PUB_KEY_OID_OFFSET(cp), IdEcPublicKey, sizeof(IdEcPublicKey))) {
+        return (int)PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    if (!memcmp(CURVE_TYPE_OID_OFFSET(cp), Secp256r1, sizeof(Secp256r1))) {
+        ctx->curve_byte_count = 32;
+        ctx->required_algorithm = PSA_ALG_SHA_256;
+    } else if (!memcmp(CURVE_TYPE_OID_OFFSET(p), Secp384r1, sizeof(Secp384r1))) {
+        ctx->curve_byte_count = 48;
+        ctx->required_algorithm = PSA_ALG_SHA_384;
+    } else {
+        return (int)PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    get_public_key_from_rfc5280_encoding(cp, &key_size);
+    /* Set attributes and import key */
+    psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_VERIFY_HASH);
+    psa_set_key_algorithm(&key_attributes, PSA_ALG_ECDSA(ctx->required_algorithm));
+    psa_set_key_type(&key_attributes, PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_FAMILY_SECP_R1));
+
+    return (int)psa_import_key(&key_attributes, *cp, key_size, &ctx->key_id);
+}
+
+/* Verify the signature against the provided hash. The signature gets parsed from
+ * the encoding first, then PSA Crypto has a dedicated API for ECDSA verification
+ */
+static inline int bootutil_ecdsa_verify(const bootutil_ecdsa_context *ctx,
+    uint8_t *hash, size_t hlen, uint8_t *sig, size_t slen)
```

Comment:
The function signature is wrong. The `uint8_t *pk` and `size_t pk_len` parameters are missing.

Reply:
Done. I have asked @Roolli to test the PSA Crypto abstraction with the remaining of his patches as well (both for P256 and P384)